model/textparse/promparse.go

```diff
@@ -251,6 +251,13 @@ func (p *PromParser) nextToken() token {
 	}
 }
 
+func (p *PromParser) errorPosition(err error) error {
+	str := string(p.l.b[:p.l.i])
+	line := 1 + strings.Count(str, "\n")
+	column := 1 + p.l.i - (strings.LastIndex(str, "\n") + len("\n"))
```

Comment:
When testing this out, column numbers for some errors were different than I expected. For example, take the following scrape body: `hello_world bad`. The returned error is `strconv.ParseFloat: parsing "bad": invalid syntax line 1, column 16`, the column number being 1 after the invalid number ("bad") ends, whereas I would expect it to be 13, the first character of the invalid number. I found that using `p.l.start` to calculate the column number fixes this. There are a few cases though when this new calculation is in the negatives, so we can fall back to using `p.l.i` instead in that scenario. Using the following snippet, there are some test failures in `promparse_test.go` with different column numbers that more correct. I assume the same is the case in `openmetricsparse_test.go` but I haven't tried it.

```suggestion
	lineStart := (strings.LastIndex(str, "\n") + len("\n"))
	column := 1 + p.l.start - lineStart
	if column < 1 {
		// The column number can't be less than 1, so resort to using
		// the index.
		column = 1 + p.l.i - lineStart
	}
```